File: typescript-connector/src/models/VirtualPlace.ts
// Class from DFC Business Ontology: #VirtualPlace
import { SemanticObject } from "../core/SemanticObject.js";
import { Place, type PlaceParams } from "./Place.js";

export interface VirtualPlaceParams extends PlaceParams {
  url?: string;
  websitePage?: string;
}
export class VirtualPlace extends Place {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:VirtualPlace";
  }

  url?: string;
  websitePage?: string;

  constructor(
    semanticId: string,
    params?: VirtualPlaceParams,
  ) {
    super(semanticId, params);
    this.url = params?.url;
    this.websitePage = params?.websitePage;
    this.semanticType = VirtualPlace.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:VirtualPlace:url", () => this.url);
    this.registerSemanticProperty("dfc-b:VirtualPlace:website_page", () => this.websitePage);
  }
  static {
    SemanticObject.typeRegistry.set(VirtualPlace.SEMANTIC_TYPE, VirtualPlace);
  }
}

File: ruby-gem/lib/models/virtual_place.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #VirtualPlace
require_relative '../core/semantic_object'
require_relative 'place'


module DfcLinkmlConnector
  module Models
    class VirtualPlace < Place
      SEMANTIC_TYPE = "dfc-b:VirtualPlace".freeze

      # @return [String]
      attr_accessor :url

      # @return [String]
      attr_accessor :website_page

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, hosts: nil, url: nil, websitePage: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, hosts: nil, url: nil, websitePage: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension, hosts: hosts)
        @url = url
        @website_page = websitePage
        self.semanticType = "dfc-b:VirtualPlace"
        registerSemanticProperty("dfc-b:URL", &method("url")).valueSetter = method("url=")
        registerSemanticProperty("dfc-b:websitePage", &method("website_page")).valueSetter = method("website_page=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
